### PR TITLE
Fix #1839: Add CASE_INSENSITIVE_ORDER to RuntimeString.

### LINKED ISSUE #1839: String.CASE_INSENSITIVE_ORDERING is missing

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
Review by @sjrd

**Comment by sjrd** [comment]
LGTM

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/StringTest.scala [hunk 1]
```diff
@@ -337,5 +337,18 @@ object StringTest extends JasmineTest {
...
+        String.CASE_INSENSITIVE_ORDER.compare(s1, s2)
+
+      expect(compare("a", "a")).toBe(0)
+      expect(compare("a", "A")).toBe(0)
+      expect(compare("A", "a")).toBe(0)
+      expect(compare("Scala.JS", "Scala.js")).toBe(0)
+      expect(compare("Scala.JS", "scala")).toBeGreaterThan(0)
+      expect(compare("åløb", "ÅLØB")).toBe(0)
+      expect(compare("Java", "Scala")).toBeLessThan(0)
+    }
```
Comment (gzm0) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/StringTest.scala]: A comparator returning always 0 will pass these tests. Maybe add some tests for order?

Comment (sjrd) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/StringTest.scala]: That's not true. There is `expect(compare("Scala.JS", "scala")).toBeGreaterThan(0)` and `expect(compare("Java", "Scala")).toBeLessThan(0)`

Comment (nicolasstucki) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/StringTest.scala]: I took those tests from `compareToIgnoreCase`. I will remove them from here and improve the tests of `compareToIgnoreCase`.

Comment (gzm0) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/StringTest.scala]: Nah, its all fine. I just should read code properly before complaining... Sorry for the noise.

**Comment by sjrd** [comment]
Re-LGTM

**Comment by sjrd** [comment]
Needs rebase, unfortunately.

**Comment by sjrd** [comment]
The failure is spurious, but this needs a rebase.

**Pull request merged**